components/eam/src/physics/cam/physics_utils.F90

```diff
@@ -32,6 +33,53 @@ module physics_utils
 
 #endif
 
-    contains
+contains
+
+  pure function calculate_drymmr_from_wetmmr(ncols, wetmmr, qv_wet) result (drymmr)
+    !Compute drymmr for any wetmmr constituent using wet water vapor mixing ratio(qv_wet)
+
+    use ppgrid, only: pcols, pver
+
+    implicit none
+
+    !intent-ins
+    integer,     intent(in) :: ncols       !number of columns
+    real(rtype), intent(in) :: wetmmr(:,:) !wet mmr of a constituent
+    real(rtype), intent(in) :: qv_wet(:,:) !water vapor wet mass mixing ratio
+
+    !return variable
+    real(rtype) :: drymmr(pcols,pver) !dry mmr of a constituent
+
+    !Compute drymmr
+    drymmr(:ncols,:) = wetmmr(:ncols,:)/(1.0_rtype - qv_wet(:ncols,:))
+
+    !Since pcols can be > ncols, assign uninitialized columns using "huge"
+    drymmr(ncols+1:pcols,:) = huge(1.0_rtype)
+
+  end function calculate_drymmr_from_wetmmr
+
+
+  pure function calculate_wetmmr_from_drymmr(ncols, drymmr, qv_dry) result (wetmmr)
+    !Compute wetmmr for any drymmr constituent using dry water vapor mixing ratio(qv_dry)
+    
+    use ppgrid, only: pcols, pver
+
+    implicit none
+
+    !intent-ins
+    integer,     intent(in) :: ncols      !number of columns
+    real(rtype), intent(in) :: drymmr(:,:)!dry mmr of a constituent
+    real(rtype), intent(in) :: qv_dry(:,:)!water vapor dry mass mixing ratio
+
+    !return variable
+    real(rtype) :: wetmmr(pcols,pver) !wet mmr of a constituent
+
+    !Compute wetmmr
+    wetmmr(:ncols,:) = drymmr(:ncols,:)/(1.0_rtype + qv_dry(:ncols,:))
+
+    !Since pcols can be > ncols, assign uninitialized columns using "huge"
+    wetmmr(ncols+1:pcols,:) = huge(1.0_rtype)
```

Comment:
Same as above: initialize everywhere to huge, then overwrite where we actually will use the entries.